frontend/src/components/GoalCards/__tests__/GoalsCardsHeader.js

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { render, act, screen } from '@testing-library/react';
+import { Router } from 'react-router';
+import { createMemoryHistory } from 'history';
+import userEvent from '@testing-library/user-event';
+import GoalCardsHeader from '../GoalsCardsHeader';
+import UserContext from '../../../UserContext';
+
+describe('GoalDataController', () => {
+  const DEFAULT_USER = {
+    name: '',
+    id: 1,
+    flags: ['merge_goals'],
+  };
+
+  const REGION_ID = 1;
+  const RECIPIENT_ID = 1;
+
+  const dismissMergeSuccess = jest.fn();
+
+  const defaultProps = {
+    title: 'TTA Goals',
+    count: 1,
+    activePage: 1,
+    offset: 0,
+    perPage: 10,
+    handlePageChange: jest.fn(),
+    hidePagination: true,
+    sortConfig: {
+      sortBy: 'mergedGoals',
+      direction: 'ASC',
+      activePage: 1,
+      offset: 0,
+    },
+    requestSort: jest.fn(),
+    numberOfSelectedGoals: 0,
+    allGoalsChecked: false,
+    selectAllGoalCheckboxSelect: jest.fn(),
+    selectAllGoals: jest.fn(),
+    selectedGoalIds: [],
+    perPageChange: jest.fn(),
+    pageGoalIds: 1,
+    createRttapa: jest.fn(),
+    showRttapaValidation: false,
+    draftSelectedRttapa: [],
+    shouldDisplayMergeSuccess: true,
+    dismissMergeSuccess,
+    filters: [],
+    recipientId: String(RECIPIENT_ID),
+    regionId: String(REGION_ID),
+    hasActiveGrants: true,
+    showNewGoals: false,
+    canMergeGoals: true,
+  };
+  const history = createMemoryHistory();
+
+  const renderTest = (props = {}, locationState = undefined) => {
+    history.location.state = locationState;
+
+    render(
+      <UserContext.Provider value={{ user: DEFAULT_USER }}>
+        <Router history={history}>
+          {/* eslint-disable-next-line react/jsx-props-no-spreading */}
+          <GoalCardsHeader {...defaultProps} {...props} />
+        </Router>
+      </UserContext.Provider>,
+    );
+  };
+
+  it('displays correct message with merged goals', async () => {
```

Comment:
Do we want to test with alt values for:
shouldDisplayMergeSuccess
dismissMergeSuccess

Also location undefined.

Reply:
This stuff is tested at higher levels, no need to repeat here